docs/DeveloperGuide.md

```diff
@@ -1,3 +1,1140 @@
+**Developer Guide**
+===================
+
```

Comment:
The DG is well written with many details about the functions and implementations. While the diagrams are lacking, you can still add them on later. I think the formatting of the DG can also be standardised to look neater but overall it was a well written documentation. Good job! And shoutout to MsKwek MingShun98!